src/analyses/goto_check.cpp

```diff
@@ -275,21 +275,14 @@ void goto_checkt::invalidate(const exprt &lhs)
   else if(lhs.id()==ID_symbol)
   {
     // clear all assertions about 'symbol'
-    find_symbols_sett find_symbols_set;
-    find_symbols_set.insert(to_symbol_expr(lhs).get_identifier());
+    find_symbols_sett find_symbols_set{to_symbol_expr(lhs).get_identifier()};
 
-    for(assertionst::iterator
-        it=assertions.begin();
-        it!=assertions.end();
-        ) // no it++
+    for(auto it = assertions.begin(); it != assertions.end();)
     {
-      assertionst::iterator next=it;
-      next++;
-
       if(has_symbol(*it, find_symbols_set) || has_subexpr(*it, ID_dereference))
-        assertions.erase(it);
-
-      it=next;
+        it = assertions.erase(it);
+      else
+        ++it;
     }
```

Comment:
What about using `std::remove_if` (until we can use `erase_if` with C++20)?

Reply:
Unfortunately, `std::remove_if` requires the iterator to be `MoveAssignable`. Will have to wait for this one.

Reply:
Ah, ok, too bad.